code/modules/urist/mob/spoopymobs.dm

```diff
@@ -81,91 +199,104 @@
+/mob/living/simple_animal/hostile/urist/zombie/UnarmedAttack(var/atom/A, var/proximity)
+	. = ..()
+
 	if(plague)
-		var/infectious = src.plague
-		var/regenerative = src.regen
-		var/resilience = src.maxHealth
-		if(istype(src.target, /mob/living/carbon/human))
-			var/mob/living/carbon/human/victim = target
-			if(victim.stat == DEAD)
-				src.visible_message("<span class = 'warning'> <b>[src]</b> chomps at [victim]'s brain!</span>", "<span class = 'warning'>You munch on [victim]'s brain!</span>")
-				victim.zombify()
-				return
-		else if(istype(src.target, /mob/living/simple_animal/hostile/scom/civ))
-			var/mob/living/simple_animal/hostile/scom/civ/victim = target
-			if(victim.stat == DEAD)
-				src.visible_message("<span class = 'warning'> <b>[src]</b> chomps at [victim]'s brain!</span>", "<span class = 'warning'>You munch on [victim]'s brain!</span>")
-				victim.Zombify(regenerative, infectious, resilience)
-				return
-	return ..()
-
-/mob/living/simple_animal/hostile/urist/zombie/verb/EatBrain()
-
-	set name = "Eat Brain"
-	set desc = "Eat a target corpse's brain to zombify him."
-
-	if(stat || !(plague))
-		src << "You cannot eat brains in your current state."
-		return
+		if(ishuman(A))
+			var/mob/living/carbon/human/victim = A
+
+			if(victim.reagents)
+				victim.reagents.add_reagent(/datum/reagent/toxin/zombie/uristzombie, rand(1, 15))
+				uZombieInfect(victim)
+
+		else if(istype(A, /mob/living/simple_animal/hostile/scom/civ))
+			var/mob/living/simple_animal/hostile/scom/civ/victim = A
+			uZombieInfect(victim)
 
-	var/list/choices = list()
-	for(var/mob/living/carbon/human/target in view(1,src))
-		if((target.stat == DEAD) && Adjacent(target))
-			choices += target
-	for(var/mob/living/simple_animal/hostile/scom/civ/C in view(1,src))
-		if((C.stat == DEAD) && Adjacent(C))
-			choices += C
-	var/mob/living/T = input(src,"Whose brain tissue do you wish to sample?") as null|anything in choices
-
-	if(!T || !src || src.stat) return
-
-	if(istype(T, /mob/living/carbon/human))
-		var/mob/living/carbon/human/victim = T
-		src.visible_message("<span class = 'warning'><b>[src]</b> chomps at [victim]'s brain!</span>", "<span class = 'warning'>You munch on [victim]'s brain!</span>")
-		victim.zombify()
-	else if(istype(T, /mob/living/simple_animal/hostile/scom/civ))
-		var/mob/living/simple_animal/hostile/scom/civ/victim = T
-		src.visible_message("<span class = 'warning'> <b>[src]</b> chomps at [victim]'s brain!</span>", "<span class = 'warning'>You munch on [victim]'s brain!</span>")
-		victim.Zombify(src.regen, src.plague, src.maxHealth)
 	return
 
-/*
-/mob/living/carbon/human/proc/Zombify(var/regens = 0, var/infects = 0, var/hitpoints = 40) //I swear officer, that Animalize() proc fell out the back of a truck.
+// do excuse the weird naming convention, but Bay *already* broke this proc once so I'm not taking chances.
+/mob/living/simple_animal/hostile/urist/zombie/proc/uZombieInfect(var/mob/living/infectee)
```

Comment:
This seems to ignore clothing like biosuits and sealed suits. That seems like a bad idea, since i doubt a decaying human's teeth can puncture a spaceproof suit enough to exchange a significant amount of fluids.

Reply:
If it did, it... well, did. Keep in mind, it's not your regular 'corpse except alive' zombie, this is explicitly regenerative - so much so that if it transforms the brain, the only solution is to pulverize the whole body. It just uses a lot of energy and didn't quite figure out higher brain functions yet, hence BRAAAAAINS.